### ISSUE TITLE
No warning about changed file when swiching buffer

### ISSUE THREAD

**Issue by bodograumann** [vim-ctrlspace/vim-ctrlspace#235]
**Describe the bug**
When opening an existing buffer in vim, where the associated file has been changed since the buffer was visible the last time, a warning is emitted:

> W11: Warning: File "foobar.txt" has changed since editing started
> See ":help W11" for more info.
> [O]K, (L)oad File:

This does not happen when switching buffers with vim-ctrlspace.

**To Reproduce**
1. Open foobar.txt in vim
2. Open a different buffer, e.g. with `:enew`
3. Change foobar.txt outside of vim, e.g. `echo "TEXT" >> foobar.txt`
4. Open the ctrl-space panel
5. Select foobar.txt and press enter

**Expected behavior**
A warning appears that foobar.txt has been changed outside of vim.

**Version(s) (please complete the following information):**
 - OS: gentoo linux
 - Version gvim 8.1.1486

**Additional context**
In regular vim/gvim the warning appears any time you first switch to the changed buffer. If you decide not to load the changes, the warning will not be shown until the file is changed again outside of vim.

It does not matter how you open the buffer. I tried `Ctrl-6` and `:b`.

**Comment by Konfekt** [comment]
Goo catch. I can reproduce. No clue what this behavior is due, though.

**Comment by cometsong** [comment]
I've been able to reproduce this one sometimes as well. _(Why i didn't put a comment on escapes me!)_
I'm going to re-test that, as _sometimes_ I do get the `file has changed` message. Might be only when writing to it `:w` ?

**Comment by bodograumann** [comment]
Yes, I think when writing there is a warning, @cometsong, but at that point I have already updated the wrong version. Sometimes I even redid changes that where already on the hard drive.
I also don‘t know of a good way to merge the buffer and outside changed at that point.

**Comment by Konfekt** [comment]
After all, I have to retract. There was a setting `autoread` in sensible.vim that suppressed it. After removing them, the behavior is the same with or without `vim-ctrlspace`; a warning is only triggered after a first `:write`. To trigger it by `:checktime` on reentering `Vim`, an autocmd such as

```vim
autocmd vimrc FocusGained * nested if empty(getcmdtype()) && empty(getcmdwintype()) | checktime | silent redraw! | endif
```

achieves that.

**Comment by bodograumann** [comment]
@Konfekt Ok there is one thing I forgot to mention. The `hidden` option must be set. E.g.:

```
$ echo "Test" > foobar.txt
$ vim -C -u NONE foobar.txt
    :set hidden
    :enew
    <C-z>
$ echo "Test" >> foobar.txt
$ fg
    <C-6>
```
Now the warning appears.

Not so when I run without `-C -u NONE` and at the end instead of `<C-6>` use `<C-space>j<CR>`

**Comment by Konfekt** [comment]
Thank you for the detailed instructions. On

```
VIM - Vi IMproved 8.0 (2016 Sep 12)
Patch incluse: 1-1568
Compilato da 'http://www.opensuse.org/'
```

hitting `<c-6>` produces a warning, while `<c-space>j<cr>` shows the modified file due to

```vim
autocmd vimrc FocusGained * nested if empty(getcmdtype()) && empty(getcmdwintype()) | checktime | silent redraw! | endif
```

when using `Vim` with CtrlSpace installed while `vim -C -u NONE foobar.txt` did not load `CtrlSpace`.

**Comment by bodograumann** [comment]
So do you agree that this should be changed in ctrlspace, @Konfekt ?

**Comment by Konfekt** [comment]
If there is no warning, then this better be fixed. However, `vim -C -u NONE` did not load `CtrlSpace` on my machine, and so I could not reproduce the issue.

**Comment by Konfekt** [comment]
Even after disabling all calls to `checktime` and usetting `autoread`, `foobar.txt` is updated immediately after `<C-space>j<CR>`. Perhaps someone else can reproduce?

**Comment by bodograumann** [comment]
Not loading ctrlspace was exactly the point :-D.
Running with `-u NONE` disables all plugins etc. and thus shows what the normal behaviour is and that the warning is not caused by any special setting or plugin apart from `hidden`.

I now also checked that the warning is not shown when using ctrlspace and no other plugin. For that I switched to a new linux user with only the system default settings and no .vim/.vimrc.
Then I installed ctrlspace and tried `vim` … `<C-space>j<CR>`. It did not produce any warning.

**Comment by Konfekt** [comment]
`CtrlSpace` needs `hidden` to be set. Does any of these settings
```vim
set autoread
" Check if file changed when its window is focused, more eager than 'autoread'
autocmd vimrc WinEnter * checktime
```
resolve the issue? It does not hurt to recommend these as a workaround in the meanwhile before harm is done.

**Comment by bodograumann** [comment]
Using `autoread` does load the file without a warning. So it could be used as a workaround.

Having no warning message still is a problem when the hidden buffer contains unsaved changes though. Usually the warning then reads:
> W12: Warning: File "foobar.txt" has changed and the buffer was changed in Vim as well

With `autoread` and `<C-space>j<CR>` the buffer is not changed and you only get a warning when you try to write. (Same as if you don’t reload the changed file)

The `autocmd` is not accepted by my vim:
> E216: No such group or event: vimrc WinEnter * checktime

**Comment by Konfekt** [comment]
> The autocmd is not accepted by my vim:

Sorry, it is

```vim
autocmd WinEnter * checktime
```

With a minimal `vimrc`,

```vim
set nocompatible
filetype plugin indent on
syntax on
let &rtp = '~/.vim/plugged/vim-ctrlspace' . ',' . &rtp
set hidden
```

Vim stalls when hitting `<cr>`.

**Comment by bodograumann** [comment]
Yes I can confirm that this `autocmd` leads to stalling.

**Comment by Konfekt** [comment]
The stalling already occurs with

```vim
set nocompatible
filetype plugin indent on
syntax on
let &rtp = '~/.vim/plugged/vim-ctrlspace' . ',' . &rtp
set hidden
```

**Comment by bodograumann** [comment]
Oh yeah seems like it already does with only `set hidden` and ctrlspace installed.
Though I wouldn't call it stalling anymore.
Pressing `<CR>` again opens the buffer successfully.

With `autoread` enabled, pressing `<CR>` once suffices.

**Comment by Konfekt** [comment]
It seems the problem is that CtrlSpace suppresses the warning dialog.
